fontforgeexe/charinfo.c

```diff
@@ -3653,6 +3659,69 @@ static int CI_DefLCChange(GGadget *g, GEvent *e) {
 return( true );
 }
 
+// Turn the user's entered decomposition into the format returned by SFGetAlternate
+unichar_t* CI_ParseUserDecomposition(char* inp) {
+    unsigned long len = strlen(inp);
+    char* end;
+    unichar_t* out = malloc(len*sizeof(unichar_t));
+
+    int j = 0;
+    for (unsigned long i = strtoul(inp, &end, 16); inp != end; i = strtoul(inp, &end, 16)) {
+        inp = end;
+        out[j] = i;
+        j++;
+    }
+
+    out[j] = '\0';
+
+    return out;
+}
+
+char* CI_CreateInterpretedAsLabel(unichar_t* inp) {
+    char* lblprefix = "Interpreted as: ";
+    char* lblerror = "Error: wrong format";
+    // 2 makes it large enough to hold the error string
+    int factor = inp == NULL ? 2 : u_strlen(inp);
+    char* lblbuf = malloc(strlen(lblprefix)+(factor*4));
```

Comment:
How so? The maximum UTF-8 bytes a unichar_t can produce is four, and the factor is the length of the input. In the opposite case, the factor is just enough to hold the difference between `lblerror` and `lblformat`.

Reply:
If that's the case, then aren't you missing one for the null terminator?

Anyway, why I say this is dodgy is not so much because you *can* inspect what the max will be, it's that it's a fragile approach if the assumptions change (e.g. what if someone came along and decided to change that error message), and you can (and probably did, as it is) get it wrong, vs. just computing the length and allocating after the fact.

Reply:
Another problem with this is that it's not gettext'ed, so it's not translatable, use the `_()` macro (although I forget how you use it with format variables - you'll need to look that one up)